modules/extension/xsd/xsd-gml2/src/main/java/org/geotools/gml2/simple/GeometryEncoder.java

```diff
@@ -33,10 +33,18 @@
  */
 public abstract class GeometryEncoder<T extends Geometry> extends ObjectEncoder<T> {
 
+    /** Flag for to encode gml:id as attribute if is true. */
+    private boolean encodeGmlId = true;
```

Comment:
This way the GML2 encoder also end up having the encodeGMLId property set to true (but should receive a null gmlid so the encoding does not happen), best to have it set to false and enable it in GML 3.2 instead

Reply:
We discussed internally, causes too much churn in other places, will be left as is.